### PR TITLE
[t4656] Migrating stage_discount to 9.0

### PR DESCRIPTION
**Pull request by BT-aestebanez**

### CODE REVIEW

File: stage_discount/data/default_product.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <!-- In Odoo 9, sale order lines must have a product defined, the following record is used for that lines -->
```
Comment (BT-csanchez) [inline stage_discount/data/default_product.xml]: for those lines

File: stage_discount/data/default_product.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <record id="default_product_product_stage_discount" model="product.product">
+            <field name="name">default_product_product_stage_discount</field>
+            <field name="categ_id" ref="product.product_category_all"/>
+            <field name="standard_price">0</field>
+            <field name="list_price">0</field>
+            <field name="type">service</field>
+            <field name="uom_id" ref="product.product_uom_unit"/>
+            <field name="uom_po_id" ref="product.product_uom_unit"/>
+            <field name="description">Default product for stage discount.</field>
+            <field name="active">False</field>
```
Comment (BT-csanchez) [inline stage_discount/data/default_product.xml]: active = False ??

File: stage_discount/models/account_invoice_ext.py [hunk 3]
```diff
@@ -131,21 +183,19 @@ def __action_compute(self):
...
         final_result = []
-        AccountInvoiceLine = self.env['account.invoice.line']
         StageDiscountDiscount = self.env['stage_discount.discount']
         amount_line_per_account_id_tax = {}
         for line in self.invoice_line_ids:
-            invoice_line = AccountInvoiceLine.browse(line.id)
-            if invoice_line.account_analytic_id:
-                context['analytic_account'] = invoice_line.account_analytic_id.id
+            if line.account_analytic_id:
+                context['analytic_account'] = line.account_analytic_id.id
```
Comment (BT-csanchez) [inline stage_discount/models/account_invoice_ext.py]: context for what!!

File: stage_discount/models/account_invoice_ext.py [hunk 4]
```diff
@@ -188,17 +228,20 @@ def action_apply_discount(self):
         Next we create the account.invoice.line with the flag is_discount = True.
         '''
         self.ensure_one()
-        context = self.env.context.copy()
-        if 'custom_search_line_discount' in context:
-            del context['custom_search_line_discount']
-        for line in self.invoice_line_ids:
+        # TODO: Default account and default tax have been set by Alvaro, should be modified depending on the project
+        default_account = self.env['account.account'].search([('name', '=', 'Umsatzsteuer')], limit=1)
```
Comment (BT-csanchez) [inline stage_discount/models/account_invoice_ext.py]: 'name', '=', 'Umsatzsteuer'!!! no!!!


Use ir.model.data !!!

File: stage_discount/models/account_invoice_ext.py [hunk 5]
```diff
@@ -188,17 +228,20 @@ def action_apply_discount(self):
...
+        if not default_account:
+            raise ValidationError(_("Please create at least one account with name 'Umsatzsteuer'"))
+        default_tax = self.env['account.tax'].search([('name', '=', '8.0%')], limit=1)
```
Comment (BT-csanchez) [inline stage_discount/models/account_invoice_ext.py]: Same same,.... please use default config for this issue

File: stage_discount/models/account_invoice_ext.py (hunk 3 above)
Comment (BT-aestebanez) [inline stage_discount/models/account_invoice_ext.py]: https://github.com/BT-aestebanez/BT-Discount/blob/esal1_t4656_Migrate_BT-Discount_to_version_9.0/stage_discount/models/discount.py#L162

**Comment by BT-aestebanez** [comment]
As agreed with @BT-faebi , I am going to merge both PR associated to t4656:
https://github.com/brain-tec/BT-Utils/pull/27
https://github.com/brain-tec/BT-Discount/pull/35
In order to have them available in the test server of Gattlen.

In this way, Cesar can test stage_discount in the server s36.

Regards

**Pull request merged**